x-pack/legacy/plugins/siem/public/components/page/network/network_top_countries_table/index.tsx

```diff
@@ -0,0 +1,213 @@
+        : i18n.DESTINATION_COUNTRIES;
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let updateTableActivePage: any;
+    if (type === networkModel.NetworkType.page) {
+      tableType =
+        flowTargeted === FlowTargetSourceDest.source
+          ? networkModel.NetworkTableType.topCountriesSource
+          : networkModel.NetworkTableType.topCountriesDestination;
+      updateTableActivePage = updateNetworkPageTableActivePage;
+    } else {
+      tableType =
+        flowTargeted === FlowTargetSourceDest.source
+          ? networkModel.IpDetailsTableType.topCountriesSource
+          : networkModel.IpDetailsTableType.topCountriesDestination;
+      updateTableActivePage = updateIpDetailsTableActivePage;
+    }
+
+    const field =
+      topCountriesSort.field === NetworkTopNFlowFields.bytes_out ||
+      topCountriesSort.field === NetworkTopNFlowFields.bytes_in
+        ? `node.network.${topCountriesSort.field}`
+        : `node.${flowTargeted}.${topCountriesSort.field}`;
+
+    return (
+      <PaginatedTable
+        activePage={activePage}
+        columns={getCountriesColumnsCurated(
+          indexPattern,
+          flowTargeted,
+          type,
+          NetworkTopCountriesTableId
+        )}
+        dataTestSubj={`table-${tableType}`}
+        headerCount={totalCount}
+        headerTitle={headerTitle}
+        headerUnit={i18n.UNIT(totalCount)}
+        id={id}
+        isInspect={isInspect}
+        itemsPerRow={rowItems}
+        limit={limit}
+        loading={loading}
+        loadPage={newActivePage => loadPage(newActivePage)}
+        onChange={criteria => onChange(criteria, tableType)}
+        pageOfItems={data}
+        showMorePagesIndicator={showMorePagesIndicator}
+        sorting={{ field, direction: topCountriesSort.direction }}
+        totalCount={fakeTotalCount}
+        updateActivePage={newPage =>
+          updateTableActivePage({
+            activePage: newPage,
+            tableType,
+          })
+        }
+        updateLimitPagination={newLimit =>
+          updateTopCountriesLimit({ limit: newLimit, networkType: type, tableType })
+        }
+      />
+    );
+  }
+);
+
+NetworkTopCountriesTableComponent.displayName = 'NetworkTopCountriesTableComponent';
+
+const mapStateToProps = (state: State, ownProps: OwnProps) =>
+  networkSelectors.topCountriesSelector(ownProps.flowTargeted, ownProps.type);
+
+export const NetworkTopCountriesTable = connect(
+  mapStateToProps,
+  {
+    updateTopCountriesLimit: networkActions.updateTopCountriesLimit,
+    updateTopCountriesSort: networkActions.updateTopCountriesSort,
+    updateNetworkPageTableActivePage: networkActions.updateNetworkPageTableActivePage,
+    updateIpDetailsTableActivePage: networkActions.updateIpDetailsTableActivePage,
+  }
+)(NetworkTopCountriesTableComponent);
+
+const SelectTypeItem = styled(EuiFlexItem)`
+  min-width: 180px;
+`;
+
+SelectTypeItem.displayName = 'SelectTypeItem';
```

Comment:
We've got three of these floating around now -- might be time for a common component?

https://github.com/elastic/kibana/blob/e211113629805690c2d4e1761dd5be75c83cbb81/x-pack/legacy/plugins/siem/public/components/page/network/network_top_n_flow_table/index.tsx#L209-L213

https://github.com/elastic/kibana/blob/fe3422cfc36425331f38a2f3b568bc2fbff3bfd5/x-pack/legacy/plugins/siem/public/components/page/network/flow_target_select_connected/index.tsx#L22-L26